### PR TITLE
[FIX] Match the PyG API for Node Input to the Loader

### LINKED ISSUE #3333: [FIX] Properly Support and Require a Dictionary for cugraph-PyG Sampling
Currently, the sampling interface in cugraph-pyg does not properly require, or even accept a dictionary for the list of input nodes, which violates the PyG API and will cause an error if there are multiple node types.  This is not difficult to resolve, and should be fixed prior to the 23.06 release.  <s>The SG and MG examples should also be updated accordingly.</s>

### PR DESCRIPTION
**Pull request by alexbarghi-nv**
Ensures heterogeneous graphs can be handled correctly in cugraph-pyg.  Also cleans up some technical debt and adds some key tests.
Resolves #3333

### CODE REVIEW

**Review by rlratzel — APPROVED** [review]
LGTM, just one comment about a debug print which need not hold up approval.

File: python/cugraph-pyg/cugraph_pyg/tests/mg/test_mg_cugraph_loader.py
Comment (alexbarghi-nv) [reply python/cugraph-pyg/cugraph_pyg/tests/mg/test_mg_cugraph_loader.py:69]: fixed

**Review by rlratzel — APPROVED** [review]

**Pull request merged**
